Signal/src/util/ThreadUtil.m

```diff
@@ -214,14 +335,15 @@ + (ThreadOffersAndIndicators *)ensureThreadOffersAndIndicators:(TSThread *)threa
         const int kUnreadIndicatorOfferOffset = -1;
 
         if (existingBlockOffer && !shouldHaveBlockOffer) {
+            DDLogInfo(@"Removing block offer");
             [existingBlockOffer removeWithTransaction:transaction];
         } else if (!existingBlockOffer && shouldHaveBlockOffer) {
             DDLogInfo(@"Creating block offer for unknown contact");
 
             // We want the block offer to be the first interaction in their
             // conversation's timeline, so we back-date it to slightly before
             // the first incoming message (which we know is the first message).
-            uint64_t blockOfferTimestamp = (uint64_t)((long long)firstMessage.timestamp + kBlockOfferOffset);
+            uint64_t blockOfferTimestamp = (uint64_t)((long long)firstMessage.timestampForSorting + kBlockOfferOffset);
```

Comment:
> Nice, I think this fixes an issue I had where block offer was appearing at the bottom of the thread instead of the top.

Yup it should.

> Dare we just make blockOfferTimestamp = kBlockOfferOffset? (and similarly for other dynamic interactions we want to appear at the beginning of the thread?

We could, but this is already working so I'm not going to bother now.  Let's do that if we ever create dynamic interactions that show up even if there are not incoming or outgoing messages.